atomic_reactor/plugins/pre_resolve_remote_source.py

```diff
@@ -33,11 +33,37 @@ def __init__(self, tasker, workflow, dependency_replacements=None):
         """
         :param tasker: ContainerTasker instance
         :param workflow: DockerBuildWorkflow instance
+        :param dependency_replacements: list<str>, dependencies for the cachito fetched artifact to
+        be replaced. Must be of the form pkg_manager:name:version[:new_name]
         """
         super(ResolveRemoteSourcePlugin, self).__init__(tasker, workflow)
         self._cachito_session = None
         self._osbs = None
-        self._dependency_replacements = dependency_replacements
+        self._dependency_replacements = self.parse_dependency_replacements(dependency_replacements)
+
+    def parse_dependency_replacements(self, replacement_strings):
+        """Parse dependency_replacements param and return cachito-reaady dependency replacement dict
+
+        param replacement_strings: list<str>, pkg_manager:name:version[:new_name]
+        return: list<dict>, cachito formated dependency replacements param
+        """
+        if not replacement_strings:
+            return
+
+        dependency_replacements = []
+        for dr_str in replacement_strings:
+            pkg_manager, name, version, new_name = (dr_str.split(':', 3) + [None] * 4)[:4]
```

Comment:
Couldn't just osbs-client pass this as JSON?

Reply:
I did think about dealing with it in osbs-client (the benefit would be earlier feedback), although it doesn't feel like this should be osbs-client's responsibility. In this case, I'd rather keep the current format.

If the idea would be to pass a JSON string to the CLI, it would not be much user friendly. If we go down this path, we'd need to check if rpkg people would feel OK with that as well. Unless OSBS client does handle the code above, translating the string to JSON.

Reply:
I find osbs-client the best place to replace custom CLI passed representation to JSON/dict. So we have to deal with that only in one place and everywhere else it works with standard formats. However if this may cause complications to koji-c, rhpkg/fedpkg I'm fine with parsing it in atomic-reactor.

OK json as param may be too much :)